bundles/AdminBundle/Controller/Searchadmin/SearchController.php

```diff
@@ -77,9 +77,9 @@ public function findAction(Request $request, EventDispatcherInterface $eventDisp
 
         $query = $this->filterQueryParam($allParams['query'] ?? '');
 
-        $types = explode(',', $allParams['type'] ?? '');
-        $subtypes = explode(',', $allParams['subtype'] ?? '');
-        $classnames = explode(',', $allParams['class'] ?? '');
+        $types = explode(',', preg_replace("/[^a-z,]/i", '', $allParams['type'] ?? ''));
+        $subtypes = explode(',', preg_replace("/[^a-z,]/i", '', $allParams['subtype'] ?? ''));
+        $classnames = explode(',', preg_replace("/[^a-z,]/i", '', $allParams['class'] ?? ''));
```

Comment:
```suggestion
        $classnames = explode(',', preg_replace("/[^a-z0-9_,]/i", '', $allParams['class'] ?? ''));
```
See https://github.com/pimcore/pimcore/blob/528335a4385604a2bf56a168d8b289ab1ee998ab/models/DataObject/ClassDefinition.php#L354 for valid names 4 classes 😊

Reply:
thanks for the hint :)